new provisoing and security documentation (#601)
nrfconnect/sdk-sidewalk: doc/releases_and_migration/migration_guide_v280.rst

Reviewer: This page is not visible in built documentation

.. _migration_guide_v280:

Migration guide for the v2.8.0 of the nRF Connect SDK
*****************************************************

.. contents::
   :local:
   :depth: 2

Starting with the v2.8.0 release of the `nRF Connect SDK`_, there is a new Sidewalk provisioning module.
For an overview of changes, read the following sections.

Reasons for change
==================

The new provisioning module enhances security by:

* Moving non-volatile Sidewalk keys to secure key storage.
* Protecting the manufacturing data partition from write operations.
* Always using the Key-Value Store version of manufacturing data.

> 

Reviewer 2: Just thought of it now, but maybe this (L22-41) could be a separate section called, e.g., "Implementation details", or a bit more precise: "Security enhancements details"? "Reasons for change" seem to end with the above bullet points

```suggestion

Security enhancements details
=============================
```

* Always use the Key-Value Store version of manufacturing data.

An access to the manufacturing data is unified in application - it is the same for prototyping and manufacturing flows.

> The mfg parser module launched on init will writhe the manufacturing data in flash (secure key storage) the same way on device certification module does.

Reviewer: ```suggestion
The mfg parser module launched on init will write the manufacturing data in flash (secure key storage) the same way on device certification module does.
```

It is triggered only once, at the first application start after flashing a mfg hex.
The following logs are present:

.. code-block:: console

   [00:00:00.004,502] <inf> sid_mfg: Need to parse mfg data
   [00:00:00.009,993] <inf> sid_mfg_parser_v8: MFG_ED25519 import success
   [00:00:00.014,900] <inf> sid_mfg_parser_v8: MFG_SECP_256R1 import success
   [00:00:00.054,505] <inf> sid_mfg: Successfully parsed mfg data

> The secure key storage is based on `hardware unique key<https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/libraries/security/hw_unique_key.html#lib-hw-unique-key>`_ security library.

Reviewer: link is not displayed correctly

Once the keys are imported to secure key storage it is impossible to read it's raw value.
Therefore once the secure key storage is enabled in firmware it must be enabled for all next firmware versions.
This features is enabled by default in all Sidewalk samples. 

The hardware flash write protection is based on `hardware flash write protection <https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/libraries/security/bootloader/fprotect.html>`_ security library.
The protection is done on Sidewalk initialization and is irreversible until a reset occurs.

> This feature is disable is Sidewalk DUT sample to alow writing manufacturing data using on device certification commands.

Reviewer 3: disable **in** Sidewalk DUT ... 

just a typo

==========================================

The new provisioning module implementation includes the following changes:

* In the :file:`app_mfg_config.h` file, the function ``app_mfg_cfg_is_valid()`` was renamed to ``app_mfg_cfg_is_empty()``.

* The ``CONFIG_SIDEWALK_MFG_STORAGE_SUPPORT_HEX_v7`` Kconfig option adds support for older Sidewalk manufacturing HEX formats (version 7 and below). 
  You can disable this configuration when using the latest manufacturing flash.

> 

Reviewer 2: ```suggestion
```

Using old provisioning module
=============================

> Although it is highly recommended to use the new provisioning module implementation, the old one can be enabled using ``CONFIG_DEPRECATED_SIDEWALK_MFG_STORAGE`` configuration. 

Reviewer 2: ```suggestion
It is recommended to use the new provisioning module implementation. 
However, you can still enable the old module using the ``CONFIG_DEPRECATED_SIDEWALK_MFG_STORAGE`` Kconfig option. 
```
